datumaro/components/explorer.py

```diff
@@ -43,22 +51,29 @@ def __init__(
         """
         self._model = ExplorerLauncher(model_name="clip_visual_ViT-B_32")
         self._text_model = ExplorerLauncher(model_name="clip_text_ViT-B_32")
-        inference = dataset.run_model(self._model, append_annotation=True)
         self._topk = topk
-
         database_keys = []
         item_list = []
 
-        for item in inference:
-            for annotation in item.annotations:
-                if isinstance(annotation, HashKey):
-                    try:
-                        hash_key = annotation.hash_key[0]
-                        hash_key = np.unpackbits(hash_key, axis=-1)
-                        database_keys.append(hash_key)
-                        item_list.append(item)
-                    except Exception:
-                        hash_key = None
+        datasets = datasets[0]
+        if isinstance(datasets, IDataset):
+            datasets = [datasets]
+
+        for dataset in datasets:
+            uninferenced_dataset = select_uninferenced_dataset(dataset)
```

Comment:
```suggestion
    datasets_to_infer = [select_uninferenced_dataset(dataset) for dataset in datasets]
    datasets = self.compute_hash_key(datasets, datasets_to_infer)

def compute_hash_key(self, datasets, datasets_to_infer):
      for dataset in datasets_to_infer:
            if len(dataset) > 0:
                  dataset.run_model(self.model, append_annotation)

        for dataset, dataset_to_infer in zip(datasets, datasets_to_infer):
             dataset.update(dataset_to_infer)
        return datasets

@property
def model(self):
     if self._model is None:
           self._model = ExplorerLauncher(model_name="clip_visual_ViT-B_32")
     return self._model
```

Allow lazy loading for the model launcher when there are no model inference requirements.